src/pages/Account/ProfilView.jsx

```diff
@@ -74,24 +102,387 @@ export function ProfilView() {
     setBooks(newBooks);
   };
 
+  const formattedDate = () => {
+    const dt = new DateTime(userData.creation_date)
+    return dt.toLocaleString(DateTime.DATE_MED);
+  };
+
+  const updateProfile = () => {
+    setClickOnUpdate(true);
+  };
+
+  const saveProfile = () => {
+    handleUpdateUser().then((updateFormIsValid) => {
+      if(!updateFormIsValid) {
+        return;
+      }
+      toast.success('Profile updated', {
+        position: 'bottom-right',
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: 'light',
+        transition: Bounce
+      });
+      setClickOnUpdate(false);
+    })
+
+  };
+
+  const handleUpdateUser = async () => {
+    const classError = 'profile-general-container-content-form-input-error';
+    setErrorMessageForm('');
+    
+    const firstnameInput = document.getElementById('firstname');
+    const lastnameInput = document.getElementById('lastname');
+    const emailInput = document.getElementById('email');
+    const passwordInput = document.getElementById('password');
+    const confirmPasswordInput = document.getElementById('confirm-password');
+    
+    firstnameInput.classList.remove(classError);
+    lastnameInput.classList.remove(classError);
+    emailInput.classList.remove(classError);
+    passwordInput.classList.remove(classError);
+    confirmPasswordInput.classList.remove(classError);
+
+    if(firstname === '') {
+      firstnameInput.classList.add(classError);
+      setErrorMessageForm('Firstname must not be empty');
+      return false;
+    }
+
+    if(lastname === '') {
+      lastnameInput.classList.add(classError);
+      setErrorMessageForm('Lastname must not be empty');
+      return false;
+    }
+
+    if(email === '') {
+      emailInput.classList.add(classError);
+      setErrorMessageForm('Email must not be empty');
+      return false;
+    }
```

Comment:
oue donc une methode qui prend en params (inputType) ferait la même pour tout ça non ? 

genre const handleInputEmptyError(type, input) => {
   input.classList.add(classError);
   setErrorMessageForm(`${type} must not be empty`)
   }